File: atcoder.jp/contests/abc287/tasks/abc287_b/main.rb
# frozen_string_literal: true

require 'set'

N, M = gets.split.map(&:to_i)
S = Array.new(N) { gets.chomp }
T = Array.new(M) { gets.chomp }

st = Set.new(T)
ans = S.count { |s| st.include? s.slice(3, 3) }

p ans

File: atcoder.jp/contests/abc287/tasks/abc287_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const [N, M] = input().split(" ").map(Number);
const S = [...Array(N)].map(() => input());
const T = [...Array(M)].map(() => input());

const st = new Set(T);
const ans = S.filter((s) => st.has(s.slice(3))).length;

console.log(ans);
